src/DependencyInjection/Compiler/GlobalVariablesCompilerPass.php

```diff
@@ -26,7 +26,12 @@ class GlobalVariablesCompilerPass implements CompilerPassInterface
 {
     public function process(ContainerBuilder $container)
     {
+        if (!$container->hasDefinition('twig')) {
+            return;
+        }
+
         $container->getDefinition('twig')
-            ->addMethodCall('addGlobal', ['sonata_page', new Reference('sonata.page.twig.global')]);
+            ->addMethodCall('addGlobal', ['sonata_page', new Reference('sonata.page.twig.global')])
+            ->addMethodCall('addGlobal', ['sonata_page_admin', new Reference('sonata.page.admin.page')]);
```

Comment:
The `SonataAdminBundle` is an optional dependency, so this can be null @VincentLanglet

Reply:
Why is it optional ? It's in the require
https://github.com/sonata-project/SonataPageBundle/blob/3.x/composer.json#L28

Reply:
cc @franmomu

Reply:
Fixed in https://github.com/sonata-project/SonataPageBundle/pull/1310, the services are loaded if `SonataAdminBundle` is available:

https://github.com/sonata-project/SonataPageBundle/blob/654fc11cffe2131657cf6e26d84c7a5adf32d344/src/DependencyInjection/SonataPageExtension.php#L77-L85

but I guess another PR is needed to make it really optional.